File: Ejercicio 10/transportePublico.py
from abc import ABC,abstractmethod

class Transportar(ABC):
    @abstractmethod
    def transportarPersonas(self): pass

class TransportePublico:
    def __init__(self,tipo):
        self.tipo=tipo

class Bus(TransportePublico,Transportar):
    def __init__(self): super().__init__("Bus")
    def transportarPersonas(self):
        print("Bus transportando 50 pasajeros")

class Metro(TransportePublico,Transportar):
    def __init__(self): super().__init__("Metro")
    def transportarPersonas(self):
        print("Metro transportando 100 pasajeros")

class Taxi(TransportePublico,Transportar):
    def __init__(self): super().__init__("Taxi")
    def transportarPersonas(self):
        print("Taxi transportando 4 pasajeros")

lista=[Bus(),Metro(),Taxi()]

for t in lista:
    t.transportarPersonas()

File: Ejercicio 10/transportePublico.java
interface Transportar{
    void transportarPersonas();
}

abstract class TransportePublico{
    String tipo;

    TransportePublico(String t){ tipo=t; }
}

class Bus extends TransportePublico implements Transportar{
    Bus(){ super("Bus"); }

    public void transportarPersonas(){
        System.out.println("Bus transportando 50 pasajeros");
    }
}

class Metro extends TransportePublico implements Transportar{
    Metro(){ super("Metro"); }

    public void transportarPersonas(){
        System.out.println("Metro transportando 100 pasajeros");
    }
}

class Taxi extends TransportePublico implements Transportar{
    Taxi(){ super("Taxi"); }

    public void transportarPersonas(){
        System.out.println("Taxi transportando 4 pasajeros");
    }
}

public class Main{
    public static void main(String[] args){

        Transportar[] lista={
            new Bus(),
            new Metro(),
            new Taxi()
        };

        for(Transportar t:lista)
            t.transportarPersonas();
    }
}
